### PR TITLE
Case Grouping Report : Downgrade max cases in geohash bucket from 10k to 1k

### PR DESCRIPTION
**Pull request by ajeety4**
## Product Description

The case grouping reports at present supports 10k as max cases in a geohash bucket. This causes performance issues in browser i(extreme lag and page crash) while interacting with mapbox markers for a bucket with such no. of cases.
The change made downgrades this value to 1k where the performance is way better and working with markers is quite smooth.
The value 1k was selected as a conservative estimate considering mapbox does not recommends to use markers for data points with thousands in numbers.

Note for future:  if say need arise to have a larger bucket , we should consider to use [layers](https://docs.mapbox.com/mapbox-gl-js/example/geojson-markers/) for data points instead of markers.

## Technical Summary

[Ticket](https://dimagi.atlassian.net/browse/SC-3655)

## Feature Flag

Geospatial (Allows access to GIS functionality)

### Safety story

Extremely safe - only changes a constant value that determine max. no of cases in a bucket.
Local testing done

### Automated test coverage

Updated.

### QA Plan

None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/geospatial/const.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 ALGO_AES = 'aes'
 
 # Max number of cases per geohash
-MAX_GEOHASH_DOC_COUNT = 10_000
+MAX_GEOHASH_DOC_COUNT = 10_00
```
Comment (Charl1996) [inline corehq/apps/geospatial/const.py:7]: Maybe make this
```suggestion
MAX_GEOHASH_DOC_COUNT = 1_000
```

File: corehq/apps/geospatial/tests/test_es.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ def fake_get_max_doc_count(query, case_property, precision):
     return 2 * x + 9_983
 
 
+@patch('corehq.apps.geospatial.es.MAX_GEOHASH_DOC_COUNT', 10000)
```
Comment (Charl1996) [inline corehq/apps/geospatial/tests/test_es.py:28]: Should this be 10000 or 1000?

Comment (ajeety4) [reply corehq/apps/geospatial/tests/test_es.py:28]: Any of that should work inpractice. 
I added 10000 as the test case originally was for 10000 and the faker `fake_get_max_doc_count` uses value close to 10000.

File: corehq/apps/geospatial/const.py (hunk 1 above)
Comment (ajeety4) [reply corehq/apps/geospatial/const.py]: Addressed in https://github.com/dimagi/commcare-hq/pull/34659/commits/8095702af8a7f672cfc5e39e9bf12fe67a69b42d

**Review by Charl1996 — APPROVED** [review]

**Review by zandre-eng — APPROVED** [review]

**Pull request merged**